src/braket_container.py

```diff
@@ -254,41 +225,74 @@ def install_additional_requirements() -> None:
         log_failure_and_exit(f"Unable to install requirements.\nException: {e}")
 
 
-def run_customer_code_as_process(entry_point: str) -> int:
+def extract_customer_method(entry_point: str) -> Callable:
+    """
+    Converts entry point to a runnable function.
     """
-    When provided the name of the package and the method to run, we run them as a process.
+    if entry_point.find(":") >= 0:
+        str_module, _, str_method = entry_point.partition(":")
+        customer_module = importlib.import_module(str_module)
+        customer_method = getattr(customer_module, str_method)
+    else:
+        def customer_method():
+            return runpy.run_module(entry_point, run_name="__main__")
```

Comment:
Why did you choose runpy rather than subprocess?

Reply:
runpy runs the module in the same process, which allows us to wrap it to natively catch the exception with a try/except block. We run this wrapped method in a process with the same codepath as for function entry points, simplifying the code complexity